pkg/elle/rw_register/rw_register_test.go

```diff
@@ -0,0 +1,947 @@
+	g9.Link(core.Vertex{Value: history9[3]}, core.Vertex{Value: history9[4]}, core.Realtime)
+	g9.Link(core.Vertex{Value: history9[5]}, core.Vertex{Value: history9[6]}, core.Realtime)
+	r9 := transactionGraph2VersionGraphs(core.Realtime, history9, g9)
+	require.Equal(t, expect9, r9)
+
+	// see through failed/crashed ops
+	history10 := core.History{
+		MustParseOp("wx1"),
+		MustParseOp("rx_").WithType(core.OpTypeInfo),
+		MustParseOp("rx3"),
+		MustParseOp("wy1"),
+		MustParseOp("wy2").WithType(core.OpTypeFail),
+		MustParseOp("ry3"),
+	}
+	expect10x := core.NewDirectedGraph()
+	expect10y := core.NewDirectedGraph()
+	expect10x.Link(core.Vertex{Value: NewInt(1)}, core.Vertex{Value: NewInt(3)}, core.Realtime)
+	expect10y.Link(core.Vertex{Value: NewInt(1)}, core.Vertex{Value: NewInt(3)}, core.Realtime)
+	expect10 := map[string]*core.DirectedGraph{
+		"x": expect10x,
+		"y": expect10y,
+	}
+	g10 := core.NewDirectedGraph()
+	g10.LinkToAll(core.Vertex{Value: history10[0]}, []core.Vertex{
+		{Value: history10[1]},
+		{Value: history10[2]},
+	}, core.Realtime)
+	g10.LinkToAll(core.Vertex{Value: history10[3]}, []core.Vertex{
+		{Value: history10[4]},
+		{Value: history10[5]},
+	}, core.Realtime)
+	r10 := transactionGraph2VersionGraphs(core.Realtime, history10, g10)
+	require.Equal(t, expect10, r10)
+
+	// see through seq. failed/crashed ops
+	history11 := core.History{
+		MustParseOp("wx1"),
+		MustParseOp("rx_").WithType(core.OpTypeInfo),
+		MustParseOp("wx2").WithType(core.OpTypeFail),
+		MustParseOp("wx3"),
+	}
+	expect11x := core.NewDirectedGraph()
+	expect11x.Link(core.Vertex{Value: NewInt(1)}, core.Vertex{Value: NewInt(3)}, core.Realtime)
+	expect11 := map[string]*core.DirectedGraph{
+		"x": expect11x,
+	}
+	g11 := core.NewDirectedGraph()
+	g11.Link(core.Vertex{Value: history11[0]}, core.Vertex{Value: history11[1]}, core.Realtime)
+	g11.Link(core.Vertex{Value: history11[1]}, core.Vertex{Value: history11[2]}, core.Realtime)
+	g11.Link(core.Vertex{Value: history11[2]}, core.Vertex{Value: history11[3]}, core.Realtime)
+	r11 := transactionGraph2VersionGraphs(core.Realtime, history11, g11)
+	require.Equal(t, expect11, r11)
+
+	// see through other keys
+	history12 := core.History{
+		MustParseOp("wx1"),
+		MustParseOp("wx2"),
+		MustParseOp("wy2"),
+		MustParseOp("wx3"),
+	}
+	expect12x := core.NewDirectedGraph()
+	expect12x.Link(core.Vertex{Value: NewInt(1)}, core.Vertex{Value: NewInt(2)}, core.Realtime)
+	expect12x.Link(core.Vertex{Value: NewInt(1)}, core.Vertex{Value: NewInt(3)}, core.Realtime)
+	expect12 := map[string]*core.DirectedGraph{
+		"x": expect12x,
+		"y": core.NewDirectedGraph(),
+	}
+	g12 := core.NewDirectedGraph()
+	g12.Link(core.Vertex{Value: history12[0]}, core.Vertex{Value: history12[1]}, core.Realtime)
+	g12.Link(core.Vertex{Value: history12[0]}, core.Vertex{Value: history12[2]}, core.Realtime)
+	g12.Link(core.Vertex{Value: history12[2]}, core.Vertex{Value: history12[3]}, core.Realtime)
```

Comment:
The graph of this case

```
        ┌T2(wx2)
T1(wx1)─┤
        └t3(wy2)─T4(wx3)
```

`transactionGraph2VersionGraphs` see through other keys so we can infer the version graph for `x` from it:

```
  ┌2
1─┤
  └3
```

Reply:
We still don't need `ext-key-graph`, the `find` closure function will be much easier.

Reply:
OK, but memorize `find` can make it more efficient.